manual/custom_codecs/extras/README.md

```diff
@@ -0,0 +1,417 @@
+[TypeToken]: http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/reflect/TypeToken.html
+
+#### Guava
+
+[OptionalCodec][OptionalCodec_guava] transforms another codec to wrap its Java type in a Guava
+[Optional][Optional_guava].
+
+
+```java
+import com.datastax.driver.extras.codecs.guava.OptionalCodec;
+import com.google.common.base.Optional;
+import com.google.common.reflect.TypeToken;
+
+// The built-in driver codec, that maps CQL's int to Java's Integer
+TypeCodec<Integer> intCodec = TypeCodec.cint();
+
+// A codec that maps CQL's int to Guava's Optional<Integer>
+OptionalCodec<Integer> optionalIntCodec = new OptionalCodec<Integer>(intCodec);
+
+cluster.getConfiguration().getCodecRegistry()
+        .register(optionalIntCodec);
+
+TypeToken<Optional<Integer>> optionalIntToken = new TypeToken<Optional<Integer>>() {
+};
+
+// schema: create table example (id int primary key, v int);
+PreparedStatement pst = session.prepare("insert into example (id, v) values (?, ?)");
+
+Optional<Integer> v1 = Optional.absent();
+session.execute(pst.bind()
+        .setInt("id", 1)
+        .set("v", v1, optionalIntToken));
+// v1 saved as NULL
+
+Optional<Integer> v2 = Optional.of(12);
+session.execute(pst.bind()
+        .setInt("id", 2)
+        .set("v", v2, optionalIntToken));
+// v2 saved as 12
+```
+
+See the JDK8 Optional section above for explanations about [TypeToken].
+
+[OptionalCodec_guava]: http://docs.datastax.com/en/drivers/java/3.0/com/datastax/driver/extras/codecs/guava/OptionalCodec.html
+[Optional_guava]: http://docs.guava-libraries.googlecode.com/git-history/v16.0.1/javadoc/com/google/common/base/Optional.html
+
+
+### Primitive arrays
+
+By default, the driver maps CQL lists to collections of Java objects. For high-performance scenarios, and when the
+element type maps to a Java primitive, you might want to map to an array instead, to avoid unnecessary boxing and
+unboxing.
+
+[IntArrayCodec] allows you to do that for lists of integers:
+
+```java
+cluster.getConfiguration().getCodecRegistry()
+        .register(new IntArrayCodec());
+
+// schema: create table example (i int primary key, l list<int>)
+session.execute("insert into example (i, l) values (1, ?)",
+        new int[]{1, 2, 3});
+// array saved as [1,2,3]
+```
+
+Package [com.datastax.driver.extras.codecs.arrays][arrays] contains similar codecs for all primitive types.
+
+[IntArrayCodec]: http://docs.datastax.com/en/drivers/java/3.0/com/datastax/driver/extras/codecs/arrays/IntArrayCodec.html
+[arrays]: http://docs.datastax.com/en/drivers/java/3.0/com/datastax/driver/extras/codecs/arrays/package-summary.html
+
+### Abstract utilities
+
+The module provides two abstract classes that act as building blocks for your own codecs:
+
+[MappingCodec] is useful when you can easily convert your Java type to another type that already has a codec.
+For example, if you have a `Price` object that translates to a plain `Integer`, you can piggyback on the built-in
```

Comment:
You can use an integer if you have a fixed number of decimals (e.g. multiply by 100 if you only go down to cents). Anyway that's not really the point, I think the example is clear enough.